bin/compile

```diff
@@ -50,26 +50,53 @@ venv=$cache/venv
 mkdir -p $cache/pip
 python=python2.7
 PATH=$buildpack/$plat/bin:$venv/bin:$PATH
+steptxt="----->"
 
 virtualenv() {
     python "$buildpack/vendor/virtualenv-1.11.6/virtualenv.py" "$@"
 }
 
+warn() {
+    echo >&2 " !     $@"
+}
+
+step() {
+    echo "$steptxt $@"
+}
+
+start() {
+  echo -n "$steptxt $@... "
+}
+
+finished() {
+  echo "done"
+}
+
 if test -f $build/Godeps
 then
+    warn "Deprecated, old ./Godeps file found!"
+    warn "Please upgrade godeps (go get -u github.com/tools/godeps) and"
+    warn "re-save your dependencies: godep save -r ./..."
     name=$(<$build/Godeps jq -r .ImportPath)
     ver=$(<$build/Godeps jq -r .GoVersion)
-elif test -d $build/Godeps
+elif test -f $build/Godeps/Godeps.json
 then
+    step "Checking Godeps/Godeps.json file."
+    if ! jq -r . < $build/Godeps/Godeps.json > /dev/null; then
```

Comment:
The surrounding coding style is (for better or worse) to put `then` on a line by itself. I would suggest trying to maintain a consistent style.